specification/dartLangSpec.tex

```diff
@@ -9168,33 +9169,58 @@ \subsection{Symbols}
 representing the reserved word \VOID.
 
 \LMHash{}%
-Assume that the term $t$ is an identifier \id{}
-that does not start with an underscore,
-that $t$ is \VOID,
-that $t$ is a period separated sequence of identifiers
-\code{$id_1$.$id_2$.$\cdots$.$id_n$},
-or that $t$ is derived from \synt{operator}.
-Any two occurrences of \code{\#t} evaluate to the same object
-(\commentary{that is, symbol instances are canonicalized}),
-and no other symbol literals evaluate to that object,
-nor to a \code{Symbol} instance that is equal to that object
-according to the \lit{==} operator
+Assume that the term $t$ is such that
+one of the following conditions is satisfied:
+
+\begin{itemize}
+\item $t$ is an identifier that does not start with an underscore.
+\item $t$ is \VOID.
+\item $t$ is a period separated sequence of identifiers
+  \code{$id_1$.$id_2$.$\cdots$.$id_n$}.
+\item $t$ is derived from \synt{operator}.
+\end{itemize}
+
+\LMHash{}%
+Any two occurrences of \code{\#t} evaluate to the same object,
+and so does a symbol obtained as the value of a constant expression
+of the form \code{Symbol($s$)} or \code{\CONST{} Symbol($s$)},
+where $s$ is a constant expression evaluating to
+a string whose contents is the same characters as $t$.
+\commentary{That is, symbol instances are canonicalized.}
+
+\LMHash{}%
+In these cases, we say that the given symbol is a
+\IndexCustom{non-private symbol based on}{symbol!non-private, based on}
+the string $s$.
+If $o$ is an instance of \code{Symbol} obtained by
+a non-constant evaluation of an expression
+of the form \code{Symbol($e$)} or the form \code{new Symbol($e$)}
```

Comment:
I think Github lost my comment here ...
I see no case covering `const Symbol("_x")`. This case only does non-const, and the one above only does non-private identifiers.

Consider:
```latex
An invocation of the \code{Symbol} constructor, whether \code{Symbol($e$)}, \code{new Symbol($e$)} or \code{const Symbol($e$)}, where $e$ is an expression (constant if necessary) which evaluates to a string $s$, creates a non-private symbol based on $s$. All constant symbol instance creations which creates a non-private symbol based on the same string $s$, evaluate to the same such object.
\commentary{That is, constant symbols are canonicalized.}
```

Maybe move that before dealing with literals, so the sequence is "symbol constructor", "non-private literals", "private literals" instead of "non-private literals", "constructors", "private literals". (Just a hunch that it might read better).

Reply:
Good catch! Revised the section based on this.